Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -47,14 +50,26 @@ public let appReducer: Reducer<
         switch action {
         case .onAppear where !state.hasAppearedAtLeastOnce:
             state.hasAppearedAtLeastOnce = true
-            return Effect<Credentials?, EquatableError>.result {
-                Result {
-                    try environment.userDefaults.loadCurrentAccount()
-                        .flatMap(environment.credentials.loadCredentials)
-                }.mapError(EquatableError.init)
-            }
-            .catchToEffect()
-            .map(AppAction.authenticationResult)
+
+            return Effect.merge(
+                Effect<Credentials?, EquatableError>.result {
+                    Result {
+                        try environment.userDefaults.loadCurrentAccount()
+                            .flatMap(environment.credentials.loadCredentials)
+                    }.mapError(EquatableError.init)
+                }
+                .catchToEffect()
+                .map(AppAction.authenticationResult),
+
+                .fireAndForget {
+                    environment.notifications.promptForPushNotifications()
+                },
```

Comment:
Maybe we could do this when the user starts their first conversation?
It's always annoying to have to answer all those popups when launching an app for the first time…

Reply:
I know, although I find it less of a problem on macOS where you can still use the app. In Crisp there's a banner that you have to tap to trigger the request, so I'd expect that we would do something similar. I wouldn't view this as a final thing but something for our current use instead.